core/src/main/java/org/apache/iceberg/util/PropertyUtil.java

```diff
@@ -70,4 +73,26 @@ public static String propertyAsString(Map<String, String> properties,
     }
     return defaultValue;
   }
+
+  /**
+   * Returns subset of provided map with keys matching the provided prefix. Matching is case-sensitive and the matching
```

Comment:
We recently took this method in `master` to extract a map having s3 write tags prefix (https://github.com/apache/iceberg/pull/4259).